db/handle_buf.c

```diff
@@ -991,12 +990,23 @@ int handle_socketrequest(SBUF2 *sb, int *keepsocket, int wrongdb)
             /* This part is to avoid deadlock. */
             /* If the reply was given before the control reached here,
              * check every 1 seconds if reply was given.
-             * If the worker thread has already given signal, reply_done will be
-             * 1 and we won't go inside loop again.
-             * If the worker thread give signal, after we check for reply_done
-             * the condition will timeout after 1 secs and variable will be
+             * If the worker thread has already given signal, reply_state will
+             * be DONE and we won't go inside loop again.
+             * If the worker thread give signal, after we check for reply_state
+             * the condition will timeout after 1 sec and variable will be
              * checked again. */
-            while (p_slock->reply_done == 0) {
+            while (p_slock->reply_state != REPLY_STATE_DONE) {
+                /* Assuming here that a tag request can't be 1000 second long.
+                   We have to exit in cases of error scenarios of
+                   offloading block requests. */
+                if (num_sec > TAG_MAX_WAITS) {
```

Comment:
TAG_MAX_WAITS: turn this into a tunable with default 1000 secs?

Reply:
This is tag code. I'd rather not put too much effort into it.